src/main/java/com/uid2/admin/vertx/service/ServiceLinkService.java

```diff
@@ -118,6 +128,103 @@ private void handleServiceLinkAdd(RoutingContext rc) {
         }
     }
 
+    private void handleServiceLinkUpdate(RoutingContext rc) {
+        try {
+            siteProvider.loadContent();
+            serviceProvider.loadContent();
+
+            JsonObject body = rc.body().asJsonObject();
+            if (body == null) {
+                ResponseUtil.error(rc, 400, "json payload required but not provided");
+                return;
+            }
+            String linkId = body.getString("link_id");
+            Integer serviceId = body.getInteger("service_id");
+            Integer siteId = body.getInteger("site_id");
+            String name = body.getString("name");
+            if (siteId == null || serviceId == null || name == null) {
+                ResponseUtil.error(rc, 400, "required parameters: site_id, service_id, name");
+                return;
+            }
+
+            if (serviceProvider.getService(serviceId) == null) {
+                ResponseUtil.error(rc, 404, "service_id " + serviceId + " not valid");
+                return;
+            }
+
+            if (siteProvider.getSite(siteId) == null) {
+                ResponseUtil.error(rc, 404, "site_id " + siteId + " not valid");
+                return;
+            }
+
+            final List<ServiceLink> serviceLinks = this.serviceLinkProvider.getAllServiceLinks()
+                    .stream().sorted(Comparator.comparing(ServiceLink::getLinkId))
+                    .collect(Collectors.toList());
+
+            ServiceLink serviceLink = serviceLinks
+                    .stream().filter(s -> s.getServiceId() == serviceId && s.getSiteId() == siteId && s.getName().equals(name))
+                    .findFirst()
+                    .orElse(null);
+            if (serviceLink == null) {
+                ResponseUtil.error(rc, 404, "failed to find a service_link for serviceId: " + serviceId + ", site_id: " + siteId + " and name: " + name);
+                return;
+            }
+
+            if (serviceLinks.stream().anyMatch(sl -> sl.getServiceId() == serviceId && sl.getLinkId().equals(linkId))) {
+                ResponseUtil.error(rc, 400, "service link id already exists for service_id: " + serviceId);
+                return;
+            }
+
+            if (linkId != null && !linkId.isEmpty()) {
```

Comment:
Added it to the validation of the parameters

Reply:
Can you please point me to the code where this validation is done ?

Reply:
I validate that the link_id is not null and not empty, and have updated the search to include the link_id in the search. If it is not found, it will return a 404